pkg/kube/controllers/boshdeployment/reconciler.go

```diff
@@ -83,50 +128,345 @@ func (r *ReconcileBOSHDeployment) Reconcile(request reconcile.Request) (reconcil
+	// Create temp manifest as variable interpolation job input
+	// retrieve manifest
+	r.log.Debug("Resolving manifest")
+	manifest, err := r.resolver.ResolveManifest(instance.Spec, instance.GetNamespace())
+	if err != nil {
+		r.recorder.Event(instance, corev1.EventTypeWarning, "ResolveManifest Error", err.Error())
+		r.log.Errorf("Error resolving the manifest %s: %s", instance.GetName(), err)
+		return nil, err
+	}
+
+	instance.Status.State = opsAppliedState
+
+	return manifest, nil
+}
+
+// generateVariables converts a Manifest into kube resources and create ExtendedSecret after apply ops
+func (r *ReconcileBOSHDeployment) generateVariables(ctx context.Context, instance *bdc.BOSHDeployment, manifest *bdm.Manifest, kubeConfig *bdm.KubeConfig) error {
+	r.log.Debug("Creating variables extendedSecrets")
+	var err error
+	for _, variable := range kubeConfig.Variables {
+		// Set BOSHDeployment instance as the owner and controller
+		if err := r.setReference(instance, &variable, r.scheme); err != nil {
+			r.recorder.Event(instance, corev1.EventTypeWarning, "NewExtendedStatefulSetForDeployment Error", err.Error())
+			return errors.Wrap(err, "couldn't set reference for an ExtendedStatefulSet for a BOSH Deployment")
+		}
+
+		foundSecret := &esv1.ExtendedSecret{}
+		err = r.client.Get(ctx, types.NamespacedName{Name: variable.GetName(), Namespace: variable.GetNamespace()}, foundSecret)
+		if apierrors.IsNotFound(err) {
+			err = r.client.Create(ctx, &variable)
+			if err != nil {
+				return errors.Wrapf(err, "could not create ExtendedSecret %s", variable.GetName())
+			}
+		} else {
+			foundSecret.Spec = variable.Spec
+			err = r.client.Update(ctx, foundSecret)
+			if err != nil {
+				return errors.Wrapf(err, "could not update ExtendedSecret %s", variable.GetName())
+			}
+		}
+	}
+
+	instance.Status.State = variableGeneratedState
+
+	return nil
+}
+
+// createVariableInterpolationExJob create temp manifest and variable interpolation exJob
+func (r *ReconcileBOSHDeployment) createVariableInterpolationExJob(ctx context.Context, instance *bdc.BOSHDeployment, manifest *bdm.Manifest, variables []esv1.ExtendedSecret, jobLabels map[string]string, secretLabels map[string]string) error {
+	// Create temp manifest as variable interpolation job input
+	tempManifestBytes, err := yaml.Marshal(manifest)
+	if err != nil {
+		return errors.Wrap(err, "could not marshal temp manifest")
+	}
+
+	tempManifestSecret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      manifest.GenerateSecretName(manifest.Name),
+			Namespace: instance.GetNamespace(),
+		},
+		StringData: map[string]string{
+			"manifest.yaml": string(tempManifestBytes),
+		},
+	}
+
+	foundSecret := &corev1.Secret{}
+	err = r.client.Get(ctx, types.NamespacedName{Name: tempManifestSecret.GetName(), Namespace: tempManifestSecret.GetNamespace()}, foundSecret)
+	if apierrors.IsNotFound(err) {
```

Comment:
can you please explain what this `tempManifestSecret` is used for?

Reply:
As interpolation job input, this manifest has been already applied ops files.

Reply:
👍 
Makes sense - please add a comment with that information